BinaryTreeLevelOrderTraversal.java

```diff
@@ -0,0 +1,70 @@
+import java.io.*;
+import java.util.*;
+
+public class TreeNode {
+    int val;
+    TreeNode left;
+    TreeNode right;
+    TreeNode() {}
+    TreeNode(int val) { this.val = val; }
+    TreeNode(int val, TreeNode left, TreeNode right) {
+        this.val = val;
+        this.left = left;
+        this.right = right;
+    }
+}
+
+class BinaryTreeLevelOrderTraversal {
+    public List<List<Integer>> levelOrder(TreeNode root) {
+        // here the idea is to use bfs approach as we are going level by level
+
+        List<List<Integer>> levels = new ArrayList<List<Integer>>();
+        if (root == null) return levels;
+
+        Queue<TreeNode> queue = new LinkedList<TreeNode>();
+        queue.add(root);
+        int level = 0;
+        while ( !queue.isEmpty() ) {
+          // start the current level
+          levels.add(new ArrayList<Integer>());

```

Comment:
Another minor thing: if you want to save reallocations of the underlying array, you could allocate the ArrayList to have queue.length() capacity now.  You'd know better how to do that in Java; I don't know how.